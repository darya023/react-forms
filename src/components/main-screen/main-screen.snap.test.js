import React from 'react';
import {render} from '@testing-library/react';
import {Router} from 'react-router';
import {createMemoryHistory} from 'history';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import MainScreen from './main-screen';

const mockStore = configureStore({});
const SendingStatus = {
  INITIAL: `INITIAL`,
  FETCHING: `FETCHING`,
  SUCCESS: `SUCCESS`,
  FAILURE: `FAILURE`,
};
const fakeStore = {
  user: [],
  sendingDataStatus: SendingStatus.FAILURE,
  authorizationStatus: null,
};
it(`MainScreen should render correctly`, () => {
  const store = mockStore(fakeStore);
  const history = createMemoryHistory();

  const {container} = render(
      <Provider store={store}>
        <Router history={history}>
          <MainScreen />
        </Router>
      </Provider>
  );
  expect(container).toMatchSnapshot();
});
