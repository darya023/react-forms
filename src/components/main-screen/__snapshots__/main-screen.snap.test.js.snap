// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <a
        class="link"
        href="/"
      >
        На главную
      </a>
    </header>
    <main>
      <div>
        <span>
          Вы можете 
        </span>
        <a
          class="link"
          href="/login"
        >
           Войти 
        </a>
        <span>
           или 
        </span>
        <a
          class="link"
          href="/signup"
        >
           Зарегистрироваться
        </a>
      </div>
      Что-то пошло не так. Попробуйте позже.
    </main>
  </div>
</div>
`;
